### PR TITLE
pkg/bgpv1: Updates getPeerConfig() Method

### LINKED ISSUE #28363: pkg/bgpv1: Add Unit Tests for getPeerConfig() Method
Currently, the `getPeerConfig()` method has no unit test coverage. Add unit test coverage that is consistent with other areas of the pkg.

### PR DESCRIPTION
**Pull request by danehans**
- Adds nil checks for Neighbor and EBGPMultihopTTL types.
- Adds unit test.

Fixes: #28363

### CODE REVIEW

File: pkg/bgpv1/gobgp/server.go [hunk 1]
```diff
@@ -265,7 +269,7 @@ func (g *GoBGPServer) getPeerConfig(ctx context.Context, n types.NeighborRequest
 	}
 
 	// Enable multi-hop for eBGP if non-zero TTL is provided
-	if g.asn != uint32(n.Neighbor.PeerASN) && *n.Neighbor.EBGPMultihopTTL > 1 {
+	if g.asn != uint32(n.Neighbor.PeerASN) && n.Neighbor.EBGPMultihopTTL != nil && *n.Neighbor.EBGPMultihopTTL > 1 {
```
Comment (ldelossa) [inline pkg/bgpv1/gobgp/server.go:272]: Hey @danehans, the commit talks of introducing a unit test, but doesn't mention why this change is necessary. Was this something the original code missed?

Comment (danehans) [reply pkg/bgpv1/gobgp/server.go:272]: @ldelossa thanks for the review. Yes, since `EBGPMultihopTTL` is a pointer, it should be checked for nil before dereferencing it. I caught this when adding one of the test cases in the PR. I will update the PR description accordingly.

**Comment by danehans** [comment]
Commit b5e58f2 rebases to try fixing k8s conformance CI flake.

**Comment by danehans** [comment]
Commit 64d56c9 rebases to fix CI flakes.

**Review by ldelossa — APPROVED** [review]
LGTM, thanks for the contribution.
